### DISCUSSION TITLE
How do you set ZBM parameters?

### DISCUSSION THREAD

**Discussion by descendingnerdliness** [zbm-dev/zfsbootmenu#437] [General]
I have just installed ZBM for the first time relying upon "Complete beginner's guide to installing ZFSBootMenu". After solving a few problems, I have ZBM booting Debian Testing on ZFS and it looks great. Thank you.
The issue is that ZBM.Timeout seems to default to -1, and sits forever waiting for a \n.
The man page explains that I need to set ZBM.Timeout=5, or something, to have it boot without further intervention. But the obscurity is in how one would do that. Is there a config file somewhere? I would really appreciate clear directions and examples. Thank you.

**Comment by zdykstra** [comment]
Please link the exact guide that you followed; ZFSBootMenu defaults to a 10 second timeout. There are too many ways to install it to say conclusively how you're getting a timeout of -1.

**Comment by ahesford** [comment]
You probably just forgot to set `bootfs` on the pool.

**Comment by descendingnerdliness** [comment]
Thanks for the replies. The Guide is here https://github.com/zbm-dev/zfsbootmenu/wiki/Complete-beginner's-guide-to-installing-ZFSBootMenu#334-start-the-new-environment-enter-any-encryption-password-and-select-your-boot-envrionment-in-zfsbootmenu.

As for `bootfs` and the other parameters, it is really unclear in the docs how they are set. But there is no mention of bootfs in the guide.

**Comment by classabbyamp** [comment]
that guide is unofficial and there is an official debian install guide on the official docs: https://docs.zfsbootmenu.org/en/v2.2.x/guides/debian.html

**Comment by descendingnerdliness** [comment]
Thanks. That guide does include the bootfs parameter, which I can probably figure out how to set with the command supplied there. But since my system boots when I press enter, what I really need to know is how to set the ZBM.Timeout parameter.

**Reply by zdykstra** [reply]
Were you able to get everything working on your system?

**Comment by ahesford** [comment]
The `bootfs` property is intrinsic to ZFS. See [zpoolprop(7)](https://man.voidlinux.org/zpoolprops#bootfs) for more details.

Parameters for ZFSBootMenu are set on its kernel command-line as documented on [zfsbootmenu.7](https://docs.zfsbootmenu.org/en/v2.2.x/man/zfsbootmenu.7.html). If you are using a pre-built UEFI executable, you can use the [zbm-kcl(8)](https://docs.zfsbootmenu.org/en/v2.2.x/man/zbm-kcl.8.html) tool to modify the command-line, as well as adjust your boot environment's command line. The tool can be fetched at https://get.zfsbootmenu.org/zbm-kcl and is a standalone utility, although it needs access to the `objcopy` executable to manipulate the UEFI bundle.

**Comment by ahesford** [comment]
As for the timeout, the default is 10 seconds. The timeout is ignored when you don't set `bootfs` because ZBM never tries to guess your intent. Set the property and the default countdown should take effect.
